code/menuui/mainhallmenu.cpp

```diff
@@ -361,6 +361,9 @@ void main_hall_do_multi_ready()
 		return;
 	}
 
+	// Switch to multi here once we're sure that a multi state will be set
+	Game_mode = GM_MULTIPLAYER;
```

Comment:
I believe that `Game_mode` is already set to multiplayer by the time you get to the main hall so I'm curious if this really does anything. Maybe there is another issue somewhere too?

Reply:
I thought about removing or changing this here, but I think the purpose is not just to set it to `GM_Multiplayer` but to also clear out anything that might be accidentally left over before starting a new game.  I'd have to look through it again to be sure, but this could also set the flag correctly if reached in a mainhall that has both single and multiplayer choices, but the pilot was in single player mode previously.